File: main-service/src/main/java/ru/pgk/main_service/features/teacher/mappers/TeacherDetailsMapper.java
package ru.pgk.main_service.features.teacher.mappers;

import org.mapstruct.Mapper;
import ru.pgk.main_service.common.mapper.Mappable;
import ru.pgk.main_service.features.department.mappers.DepartmentMapper;
import ru.pgk.main_service.features.teacher.dto.TeacherDetailsDto;
import ru.pgk.main_service.features.teacher.entities.TeacherEntity;

@Mapper(componentModel = "spring", uses = DepartmentMapper.class)
public interface TeacherDetailsMapper extends Mappable<TeacherEntity, TeacherDetailsDto> {}

File: teacher-service/src/main/kotlin/ru/pgk/teacher_service/features/mappers/TeacherDetailsMapper.kt
package ru.pgk.teacher_service.features.mappers

import org.mapstruct.Mapper
import ru.pgk.teacher_service.common.mapper.Mappable
import ru.pgk.teacher_service.features.dto.TeacherDetailsDto
import ru.pgk.teacher_service.features.entities.TeacherEntity

@Mapper(componentModel = "spring")
interface TeacherDetailsMapper: Mappable<TeacherEntity, TeacherDetailsDto>